src/evaluation/Run.jl

```diff
@@ -177,6 +190,8 @@ function update_save_run!(session::ServerSession, notebook::Notebook, cells::Arr
 
 	update_dependency_cache!(notebook)
 
+	indirectly_deactivated = disable_dependent_cells!(notebook, new)
+
 	session.options.server.disable_writing_notebook_files || save_notebook(notebook)
```

Comment:
I have to determine the indirectly deactivated cells before saving (in the original implementation it was done afterwards).
In order to avoid double work, `run_reactive!` gets the list of deactivated cells as (optional) parameter.